linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml

```diff
@@ -0,0 +1,19 @@
+<def-group oval_version="5.10">
+  <definition class="compliance" id="enable_fips_mode" version="1">
+    <metadata>
+      <title>Enable FIPS Mode</title>
+      <affected family="unix">
+        <platform>Red Hat Enterprise Linux 8</platform>
+        <platform>multi_platform_fedora</platform>
+      </affected>
+      <description>Check if FIPS mode is enabled on the system</description>
+    </metadata>
+    <criteria operator="AND">
+      <extend_definition comment="check /etc/system-fips exists" definition_ref="etc_system_fips_exists" />
+      <extend_definition comment="check sysctl crypto.fips_enabled = 1" definition_ref="sysctl_crypto_fips_enabled" />
+      <extend_definition comment="Dracut FIPS module is enabled" definition_ref="enable_dracut_fips_module" />
+      <extend_definition comment="system cryptography policy is configured" definition_ref="configure_crypto_policy" />
```

Comment:
I was wondering if `enable_fips_mode`  check should really look into the selected crypto-policy, or we should select the `configure_crypto_policy` rule in the profile instead.
But I did not know that the remediation command would also be setting FIPS crypto-policy.

I think it should be more clear in rule description.

Reply:
> I was wondering if enable_fips_mode check should really look into the selected crypto-policy, or we should select the configure_crypto_policy rule in the profile instead.

I think we are going to add rule configure_crypto_policy anyway because we are going to add rules about crypto policies for RHEL8 profiles.

I thought we should not duplicate the OVAL from rule configure_crypto_policy, so I preferred to add it as an extend_definition.

My concern was: Somebody will have a tailoring in which he removes the rule configure_crypto_policy from the profile, then he messes up the crypto policy settings, then he scans his machine, the rule enable_fips_mode will pass, but he will be in fact in inconsistent state.

> But I did not know that the remediation command would also be setting FIPS crypto-policy.

I look into that and I found the `fips-mode-setup` tool calls `update-crypto-policies --set FIPS`.

I agree it should be more clear in the rule description.

Reply:
> I look into that and I found the `fips-mode-setup` tool calls `update-crypto-policies --set FIPS`.

@jan-cerny Exactly! It sets the crypto-policies to `FIPS`.

Now I believe that OVAL definition for `enable_fips_mode` should actually look for `FIPS` policy, not what is defined in `var_system_crypto_policy`

Reply:
@yuumasato Shouldn't it be that `var_system_crypto_policy` should be set to `fips`?

Reply:
The check should be that for system FIPS mode the crypto policy must be set to FIPS.

Reply:
@t8m Yes, I agree, that's clear that the crypto policy must be set to FIPS. The discussion is now rather about the way we implement the check. We already have a rule that checks the crypto policy setting. In my patch I just reuse this rule to check if crypto policy is set to FIPS. @yuumasato is proposing a different way to achieve the same thing, but his way is not clear to me.

Reply:
@jan-cerny the purpose of `var_system_crypto_policy` is for you to be able to customize what crypto policy to check for. But in this case want it fixed to `FIPS`, the rule for `enable_fips_mode` should always check for `FIPS` mode, and not what is set in the variable.

Reply:
@jan-cerny Not sure how to implement this though.

One option is to declare `var_system_crypto_policy` within definition `enable_fips_mode`. Not sure if `var_system_crypto_policy` would pick this value, though. If this approach works it could potentially override configuration of `var_system_crypto_policy` in the Profile, which can be confusing to users.

Another option, simpler, is to add a criteria that checks if `var_system_crypto_policy` is set to `FIPS`.

Reply:
@jan-cerny we should also add a check to verify that fips is not disabled in `/etc/default/grub`

Reply:
@yuumasato Thaks for your suggestions. I'll try if they work.

Reply:
@yuumasato I forgot to update you that they work.